src/plugins/kconfig/kconfig_delegate.cpp

```diff
@@ -38,22 +39,29 @@ kdb::KeySet KconfigDelegate::getConfig (Key const & parent)
 {
 	KeySet keys{ 0, KS_END };
 
-	try
+
+	ELEKTRA_LOG_DEBUG ("Parse `%s` using the kconfig plugin", parent.getString ().c_str ());
+	std::ifstream file{ parent.getString () };
+	if (!file.is_open ())
 	{
-		ELEKTRA_LOG_DEBUG ("Parse `%s` using the kconfig plugin", parent.getString ().c_str ());
-		FileUtility fileUtility{ parent.getString () };
+		throw std::runtime_error ("Could not open the file.");
+	}
 
-		ELEKTRA_LOG_DEBUG ("The file opened successfully. Start parsing");
+	ELEKTRA_LOG_DEBUG ("The file opened successfully. Start parsing");
+	try
+	{
+		FileUtility fileUtility{ parent.getString (), file };
 		KConfigParser parser{ fileUtility, keys };
 		parser.parse (parent);
-
 		ELEKTRA_LOG_DEBUG ("Parsing finished successfully");
 	}
 	catch (KConfigParserException & e)
 	{
+		file.close ();
```

Comment:
This should not be needed, if you open the file directly in the constructor it will be automatically closed in the destructor.